VAM Evolutionary Character Creation.py

```diff
@@ -1096,14 +1137,19 @@ def weighted_random_selection(self):
         return choices
             
 
-    def get_elite_from_population(self):
-        ''' returns the fittest child appearance from the population '''
-        max_rating = float()
-        for i in range(1,POP_SIZE + 1):
-            if self.chromosome[str(i)]['rating'] > max_rating:
-                max_rating = self.chromosome[str(i)]['rating']
-                max_number = i
-        return self.chromosome[str(max_number)]['appearance']
+    def get_elites_from_population(self):
+        """ Returns the children appearances where the rating is the maximum rating that the user selected.
+            If the maximum selected rating is lower than MINIMAL_RATING_FOR_KEEP_ELITES, then no appearances are returned.
+            Returns a maximum of appearances equal to user setting 'max kept elites'.
+         """
+
+        max_selected_rating = max([chromosomeValue['rating'] for chromosomeValue in self.chromosome.values()])
+
+        if max_selected_rating < MINIMAL_RATING_FOR_KEEP_ELITES:
+            return []
+
+        return [chromosomeValue['appearance'] for chromosomeValue in self.chromosome.values()
+                if chromosomeValue['rating'] == max_selected_rating][:self.settings['max kept elites']]

```

Comment:
Ok, awesome. This is something I haven't seen before. What does the list comprehension in conjunction with the list slice [:XYZ] do? I mean, I know what the function is supposed to return, but what is going on with the lists here?

Reply:
It is a list comprehension followed by a slice. The list comprehension creates a list of appearances where the rating is equal to the max_selected_rating. Then the slice makes a new list by taking the first `self.settings['max kept elites'] ` members of the first list (or fewer, if the list is smaller than that number).

I was actually thinking of splitting it up into two statements (first the list comprehension, then the slice) for clarity. I could change that if you want.

Reply:
Ah, neat! I take you up on the offer to split it into two statements for readability if you don't mind.